### PR TITLE
Graph seeker

### PR DESCRIPTION
**Pull request by AKaudinov**
- Added ability to seek/zoom graph data on each graph on the data page, using an rc-slider rail.
   - The graph seeker works with the date filter on the top bar -- switching the date filter updates the graph seeker with ability to seek data within the selected time frame
   - The graph seeker works with overlayed graphs, both: the original graph and overlay graph are able to be zoomed through
  - This also accounts for holes -- missing data -- in the graph when the latest available date with a value is considerable amount (up to a day or more) behind the selected seek date
---

- Fixed an issue where the graph selected for overlay would overlay on top of itself and result in a double render

---

- Removed rc-slider global overrides from global.css and moved them over to `DataRangePicker.module.css`. 
   - This change allowed to re-use the rc-slider on the Data page for the graph seeker without having to rewrite functionality using another rc-slider component or a native html range element, as it was previously heavily customized for the Dashboard page
        - Updated the styles to use the css modules' `:global` selector to switch the scope of the overrides to global scope so 
        rc-slider styles can be overridden via the module itself.
        - Added an additional boolean prop to the `DataRangePicker` -- `IsIndex` -- to determine whether to apply rc-slider style overrides for the index page, or the new version of it on the data page

---

Removed the Legends from the data page, as they were a bit confusing and are still visible in the tooltip window and provide better context, there.

---

Ideally, this PR would be merged into stable only after https://github.com/reaktor/pluspool-led/pull/136 is merged, as it's based on the [next-13](https://github.com/reaktor/pluspool-led/tree/next-13) branch

### CODE REVIEW

File: components/DataRangePicker/DataRangePicker.module.css [hunk 1]
```diff
@@ -76,3 +76,187 @@
...
+    transition: color 0.1s ease-in-out;
+  }
+
+  /**
+  * This element allows us to make the draggable area
+  * of .rc-slider-handle larger without having to change .rc-slider-handle size.
+  * This makes it easier to make .rc-slider-handle visually 3px tall
+  * and it's draggable area 60px larger.
+  */
+  .customSlider :global(.rc-slider-handle):before {
```
Comment (jedahan) [inline components/DataRangePicker/DataRangePicker.module.css:174]: great comments

File: components/Graph/index.js [hunk 2]
```diff
@@ -67,20 +68,70 @@ const Graph = ({
...
+  const [seekDate, setSeekDate] = useState(graph.domain[0])
+  const [data, setData] = useState(graph.data)
+  const [overlayData, setOverlayData] = useState(overlayGraph ? overlayGraph.data : [])
+
+  // run and cut data down if the minimum date range changes
+  useEffect(() => {
+    // do not filter any data on the first run
+    if (firstRun.current) {
+      return
+    }
```
Comment (jedahan) [inline components/Graph/index.js:82]: seems like a smell to me, but don't know why. is there a way we can set the initial filter to achieve the same thing?

**Review by jedahan — APPROVED** [review]

File: components/Graph/index.js (hunk 2 above)
Comment (AKaudinov) [reply components/Graph/index.js:82]: The initial filter has the data cut down for each graph from the main graphs component.
This prevents us from cutting that same data again when each graph component mounts, thus not running those operations unnecessarily -- the hooks start listening for seeker filter changes after the graph mounts and execute only if the seeker value or date filter changes, not on initial render.

useRef is used to not trigger any re-renders and avoids re-calculations, this also allows the hook to not have the ref as part of the dependency array and cause possible infinite re-renders or additional re-renders due to an unnecessary state update, and conforms to React's exhaustive dependency rule.

**Comment by AKaudinov** [comment]
Closing this PR as Nivo graphs have been replaced with Chart.js and graph interaction has been made available without the graph seeker.

**Pull request closed without merging**